DQM/SiPixelPhase1TrackResiduals/src/SiPixelPhase1TrackResiduals.cc

```diff
@@ -33,15 +34,19 @@ void SiPixelPhase1TrackResiduals::analyze(const edm::Event& iEvent, const edm::E
 
   edm::Handle<reco::VertexCollection> vertices;
   iEvent.getByToken(offlinePrimaryVerticesToken_, vertices);
-  if (!vertices.isValid() || vertices->size() == 0) return;
-  const auto primaryVertex = vertices->at(0); 
+
+  if (ApplyVertexCut_ && (!vertices.isValid() || vertices->size() == 0)) return;
+  
+  reco::Vertex primaryVertex; 
+  if (ApplyVertexCut_) primaryVertex = vertices->at(0); 
```

Comment:
please avoid this copy.  (it looks like it is used only once, so easy to just remove and  use vertices[0] where needed